src/angular/form-elements/dropdown/dropdown.component.ts

```diff
@@ -1,28 +1,200 @@
-    @Input() options:DropdownValue[] | string[];
-    @Input() value:any;
-    @Input() disabled:boolean;
+    /**
+     * The label that will show up above the drop-down
+     */
+    @Input() label: string;
+
+    /**
+     * Option can be add by a list of IDropDownGroup, IDropDownItem objects, or a list of strings
+     */
+    //@Input() options: (IDropDownGroup|IDropDownItem|string)[];
+
+    @Input() options: IDropDownOption[];
+
+    /**
+     * Initial value the drop-down should have
+     */
+    @Input() value: any;
+
+    /**
+     * Drop-down disabled flag
+     */
+    @Input() disabled: boolean;
+
+    /**
+     * The text users will see on the drop-down header when no option was selected
+     */
     @Input() placeHolder:string;
+
+    /**
+     * Drop-down required flag
+     */
     @Input() required:boolean;
 
+
+    @Input() validate:boolean;
+
+    @Input() headless = false;
+
+    @ViewChild('dropDownWrapper') dropDownWrapper: ElementRef;
+
+    /**
+     * Selected option index
+     */
+    public selectedIndex: number;
+
+    /**
+     * Selected option label
+     */
+    public selectedLabel:string;
+
+    /**
+     * Drop-down show/hide flag. default is false (closed)
+     * @type {boolean}
+     */
+    public show = false;
+
+    /**
+     * Error flag
+     * @type {boolean}
+     */
+    public error: boolean;
+
+    /**
+     * Export DropDownOptionType enum so we can use it on the template
+     */
+    public cIDropDownOptionType = DropDownOptionType;
+
+    /**
+     * Configure unselectable option types
+     */
+    private unselectableOptions = [DropDownOptionType.Disable, DropDownOptionType.Header, DropDownOptionType.HorizontalLine];
+
+    public isGroupDesign = false;
+
+    constructor(private view:ElementRef){}
+
+    ngOnInit(): void {
+        if(this.options){
+            if(this.options.find(option => option.type === DropDownOptionType.Header)){
+                this.isGroupDesign = true;
+            }
+        }
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if(changes.value){
+            this.setValueIfExist(this.value);
+        }
+    }
+
+    public isValid(){
+        return !this.error;
+    }
+
+    /**
+     * Validate when required is enabled
+     */
+    public validateDropDown(): void{
+        if(!this.disabled && this.required && (!this.value || this.value === '')){
+            this.error = true;
+            return;
+        }
+        this.error = false;
+    }
+
+    /**
+     * Set option as selected and saves it's index on the list
+     * @param index - number
+     * @param option - IDropDownItem or string
+     */
+    public selectOption(index: number, option:any):void{
+        const unselectable = this.unselectableOptions.find(optionType => option.type === optionType);
+        if(unselectable){
+            return;
+        }
+        this.selectedIndex = index;
+        this.updateSelected(option.label || option, option.value || option);
```

Comment:
You could pass `index` to `this.updateSelected()` method rather than set the value of it here.